debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -154,6 +154,8 @@ public synchronized void start(Map<String, String> props) {
                 if (taskContext.isInitialSnapshotOnly()) {
                     logger.warn("This connector will only perform a snapshot, and will stop after that completes.");
                     readers.uponCompletion("Connector configured to only perform snapshot, and snapshot completed successfully. Connector will terminate.");
+                    // Snapshot Only: stop binlogReader in order to gracefully unregister metrics, which were registered during BinlogReader construction above
+                    binlogReader.stop();
```

Comment:
@pgoranss, per the `Reader` contract, I don't think you should have to call `stop()` if you don't call `start()`. That means that the `BinlogReader` and `SnapshotReader` are not properly adhering to their contract, because they are initializing resources (the JMX metrics, in this case) before `start()`.

One option is to just move the JMX metric registration into the `start()` method, but then when the connector does both a snapshot and binlog reading, the JMX metrics for binlog reading appear only after the snapshot completes. I think that's counterintuitive.

So perhaps a better approach is to change the `Reader` interface to introduce a method that allows the implementation to initialize some resources before `start()` is called. I've taken the liberty of prototyping this change as PR #310, and would appreciate your thoughts.

Reply:
Agreed. Will follow up on #310

Reply:
If you agree that #310 is a better approach, can you close this PR?

Reply:
Yes, closing now